rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2882,12 +3122,19 @@ rmw_get_node_names(
   Participant * participant = impl->participant;
 
   auto participant_names = participant->getParticipantNames();
-  node_names->size = participant_names.size();
-  node_names->data = static_cast<char **>(rmw_allocate(node_names->size * sizeof(char *)));
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rcutils_ret_t rcutils_ret =
+    rcutils_string_array_init(node_names, participant_names.size(), &allocator);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string_safe())
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+  }
   for (size_t i = 0; i < participant_names.size(); ++i) {
-    size_t name_length = participant_names[i].size() + 1;
-    node_names->data[i] = static_cast<char *>(rmw_allocate(name_length * sizeof(char)));
-    snprintf(node_names->data[i], name_length, "%s", participant_names[i].c_str());
+    node_names->data[i] = rcutils_strdup(participant_names[i].c_str(), allocator);
+    if (!node_names->data[i]) {
+      RMW_SET_ERROR_MSG("failed to allocate memory for node name")
+      return RMW_RET_BAD_ALLOC;
```

Comment:
See: bfa8aab I didn't use a label because this was at the end of the function anyways.